utilities/utilities.go

```diff
@@ -141,3 +149,16 @@ func Unzip(zippath string, destination string) (err error) {
 	}
 	return
 }
+
+// SafeJoin performs a filepath.Join of 'parent' and 'subdir' but returns an error
+// if the resulting path points outside of 'parent'.
+func SafeJoin(parent, subdir string) (string, error) {
+	res := filepath.Join(parent, subdir)
+	if !strings.HasSuffix(parent, string(os.PathSeparator)) {
+		parent += string(os.PathSeparator)
+	}
+	if !strings.HasPrefix(res, parent) {
+		return res, fmt.Errorf("unsafe path join: '%s' with '%s'", parent, subdir)
+	}
+	return res, nil
+}
```

Comment:
Maybe this could be a good candidate for go-paths-helper

Reply:
+1, since we added this also in https://github.com/codeclysm/extract/pull/14/files#diff-705c94be3d96b8c2131b6430f14904b30b254b30ca4ce3229aecd7db83822b15R328-R337